Tests/Builder/ListBuilderTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Tests\Builder;
+
+use Prophecy\Argument;
+use Prophecy\Prophet;
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Guesser\TypeGuesserInterface;
+use Sonata\DoctrineORMAdminBundle\Admin\FieldDescription;
+use Sonata\DoctrineORMAdminBundle\Builder\ListBuilder;
+use Sonata\DoctrineORMAdminBundle\Model\ModelManager;
+use Symfony\Component\Form\Guess\Guess;
+use Symfony\Component\Form\Guess\TypeGuess;
+
+/**
+ * @author Andrew Mor-Yaroslavtsev <andrejs@gmail.com>
+ */
+final class ListBuilderTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var TypeGuesserInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $typeGuesser;
+
+    /**
+     * @var ListBuilder
+     */
+    protected $listBuilder;
+
+    /**
+     * @var AdminInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $admin;
```

Comment:
If you make the class `final`, then you should probably use `private` too. Note that we don't make test class final usually, because they are not considered part of our public API.